docs/en/index.md

```diff
@@ -159,33 +159,190 @@ Nested GridFields
 The `GridFieldNestedForm` component allows you to nest GridFields in the UI. It can be used with `DataObject` subclasses
 with the `Hierarchy` extension, or by specifying the relation used for nesting.
 
+Here is a small example of basic use of Nested GridField.
+
+```php
+class MyAdmin extends ModelAdmin
+{
+    ...
+
+    public function getEditForm($id = null, $fields = null)
+    {
+        $form = parent::getEditForm($id, $fields);
+
+        $grid = $form->Fields()->dataFieldByName(MyObject::class);
+        // Add Nested GridField to main GridField
+        $grid->getConfig()->addComponent(GridFieldNestedForm::create());
+
+        return $form;
+    }
+}
+```
+
+There are several ways to use Nested GridField. The implementation depends on the type of data you will be using in your GridField.
+For instance, if there is a DataObject that has the [`Hierarchy`](api:SilverStripe\ORM\Hierarchy\Hierarchy) extension, you can use the following approach.
+
+As an example, here we can use a typical hierarchy of Security Groups, where another Group can serve as a parent.
+
+```php
+class MySecurityAdminExtension extends Extension
+{
+    public function updateGridFieldConfig($config)
+    {
+        if ($this->owner->getModelClass() === Group::class) {
+            $config->addComponent(GridFieldNestedForm::create());
+        }
+    }
+}
+```
+
+Or also view a list of all members of a given group.
+
+```php
+class MySecurityAdminExtension extends Extension
+{
+    public function updateGridFieldConfig($config)
+    {
+        if ($this->owner->getModelClass() === Group::class) {
+            $config->addComponent(GridFieldNestedForm::create()->setRelationName('Members'));
+        }
+    }
+}
+```
+
+```yml
+SilverStripe\Admin\SecurityAdmin:
+  extensions:
+    - MyApp\MyExtensions\MySecurityAdminExtension
+```
+
+Another way to use Nested GridField together with DataObjects that do not have the `Hierarchy` extension but have `has_many` relationships with other objects.
+Let's say you have the following DataObject that has multiple levels of relationships, and an Admin section where the data of this object will be displayed.
+You want the user to be able to view information regarding this object and all nested objects on the same page, without the need to navigate to each object individually.
+In this case, you can use the following approach.
+
```

Comment:
I added a lot of code snippets below since it's not quite obvious which class should have Nested GridField.